### PR TITLE
gh-116622: Test updates for Android

### PR DESCRIPTION
**Pull request by mhsmith**
This PR fixes various Android test issues which I've discovered over the last few weeks.

* Issue: gh-116622

### CODE REVIEW

File: Lib/test/_test_multiprocessing.py [hunk 1]
```diff
@@ -4662,7 +4662,7 @@ def make_finalizers():
         old_interval = sys.getswitchinterval()
         old_threshold = gc.get_threshold()
         try:
-            sys.setswitchinterval(1e-6)
+            support.setswitchinterval(1e-6)
```
Comment (mhsmith) [inline Lib/test/_test_multiprocessing.py:4665]: I've used this wrapper function in all places which set a short switch interval. It works around a deadlock which happens on old Android emulators.

File: Lib/test/test_fcntl.py [hunk 2]
```diff
@@ -129,9 +129,9 @@ def test_fcntl_bad_file_overflow(self):
             fcntl.fcntl(BadFile(INT_MIN - 1), fcntl.F_SETFL, os.O_NONBLOCK)
 
     @unittest.skipIf(
-        platform.machine().startswith(("arm", "aarch"))
-        and platform.system() in ("Linux", "Android"),
-        "ARM Linux returns EINVAL for F_NOTIFY DN_MULTISHOT")
+        (platform.machine().startswith("arm") and platform.system() == "Linux")
+        or platform.system() == "Android",
+        "this platform returns EINVAL for F_NOTIFY DN_MULTISHOT")
     def test_fcntl_64_bit(self):
```
Comment (mhsmith) [inline Lib/test/test_fcntl.py:135]: It turns out this test works fine on Linux aarch64, but it doesn't work on Android on any architecture. So I've reverted the Linux part of the condition to the state it was in before #115918.

File: Lib/test/test_math.py [hunk 3]
```diff
@@ -2691,12 +2691,12 @@ def test_fma_infinities(self):
...
     # properly: it doesn't use the right sign when the result is zero.
-    @unittest.skipIf(support.is_wasi,
-                     "WASI fma() doesn't implement IEE 754-2008 properly")
-    @unittest.skipIf(sys.platform.startswith('freebsd'),
-                     "FreeBSD fma() doesn't implement IEE 754-2008 properly")
+    @unittest.skipIf(
+        sys.platform.startswith(("freebsd", "wasi"))
+        or (sys.platform == "android" and platform.machine() == "x86_64"),
+        f"this platform doesn't implement IEE 754-2008 properly")
     def test_fma_zero_result(self):
```
Comment (mhsmith) [inline Lib/test/test_math.py:2700]: See discussion at https://github.com/python/cpython/pull/116667#issuecomment-2018049955.

**Comment by mhsmith** [comment]
@erlend-aasland: Are you able to review this PR? It's a continuation of the previous Android test updates from #115918.

File: Lib/test/test_os.py [hunk 4]
```diff
@@ -3931,7 +3931,12 @@ def test_does_not_crash(self):
         try:
             size = os.get_terminal_size()
         except OSError as e:
-            if sys.platform == "win32" or e.errno in (errno.EINVAL, errno.ENOTTY):
+            known_errnos = [errno.EINVAL, errno.ENOTTY]
+            if sys.platform == "android":
+                # The Android testbed redirects the native stdout to a pipe,
+                # which returns a different error code.
+                known_errnos.append(errno.EACCES)
+            if (sys.platform == "win32") or (e.errno in known_errnos):
```
Comment (erlend-aasland) [inline Lib/test/test_os.py]: The parentheses are not needed.

**Review by erlend-aasland — APPROVED** [review]
LGTM, with some nits. BTW, last time I checked, the `sys.platform.startswith` idiom was still recommended practice. I'd prefer if we could follow that.

**Comment by mhsmith** [comment]
> The parentheses are not needed.

Fixed.

> last time I checked, the `sys.platform.startswith` idiom was still recommended practice

I actually updated the `sys.platform` documentation in https://github.com/python/cpython/pull/116215#issuecomment-1978826528 to limit this recommendation to FreeBSD. Ever since Python 3.3, all other PEP 11-supported platforms have  returned a fixed string without a version number.

**Comment by erlend-aasland** [comment]
@mhsmith, thanks for the heads-up, I missed that docs update (and I've must have looked at the 3.12 docs). Perhaps we should consider backporting that doc patch.

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**